pandas/core/frame.py

```diff
@@ -5444,6 +5444,34 @@ def shift(
     ) -> DataFrame:
         axis = self._get_axis_number(axis)
 
+        # Handle the case of multiple shifts
+        if is_list_like(periods):
+
+            new_df = DataFrame()
+
+            from pandas.core.reshape.concat import concat
+
+            for i in periods:
+                if not isinstance(i, int):
+                    raise TypeError(
+                        f"Value {i} in periods is not an integer, expected an integer"
+                    )
+
+                new_df = concat(
```

Comment:
this is terribly inefficient 

append to a list and concat at the end

Reply:
Got it. This would be a list of DataFrames, correct?

Reply:
yep

Reply:
Thank you!

Reply:
We now append to a list in the for loop

`new_df_list.append(
                        super()
                        .shift(periods=i, freq=freq, axis=axis, fill_value=fill_value)
                        .add_suffix(f"_{i}" if suffix is None else suffix) 
                    )`

and we concat at the end

`new_df = concat(new_df_list, axis=1)`